### DISCUSSION TITLE
UX: compact control panel menu

### DISCUSSION THREAD

**Discussion by jarrettv** [zwave-js/zwave-js-ui#461] [Ideas]
I propose a more compact control panel menu so there is less scrolling for content:

**BEFORE**
![image](https://user-images.githubusercontent.com/219295/106945688-baab4180-66ed-11eb-9aec-44daadfe682e.png)

**AFTER**
![image](https://user-images.githubusercontent.com/219295/106943182-b6315980-66ea-11eb-9729-76eb01f41ed2.png)

Mobile view:
![image](https://user-images.githubusercontent.com/219295/106942834-40c58900-66ea-11eb-8b04-6921b2011ef8.png)

This is an early mockup to get some feedback.

**Comment by ahochsteger** [comment]
@jarrettv looks really good - I like it :-)

**Comment by robertsLando** [comment]
I like it @jarrettv ! But where are the zwaveHex id lib version etc? Them should be showed somewhere

**Reply by jarrettv** [reply]
i was hoping we could relocate to sidebar or settings screen

**Reply by starman2k01** [reply]
Sorry for newb question but how do I access the control panel? Documents don't point this out...

**Reply by robertsLando** [reply]
@jarrettv Controller status must be there IMO, the others could be shown in an info popup/menu, colud also be on topbar, like an `info` icon that on hover shows those informations

**Reply by jarrettv** [reply]
Like this?
![image](https://user-images.githubusercontent.com/219295/107108997-31d1fa00-6802-11eb-82e7-a60b18d5a0a9.png)

**Reply by robertsLando** [reply]
@jarrettv yep that's good! Only the controller status btw needs to be placed somewhere, I would leave it there for now

**Comment by jarrettv** [comment]
The table was rather complicated and is not mobile friendly so I'm experimenting with a grid layout:
![image](https://user-images.githubusercontent.com/219295/107168232-ad59b580-6980-11eb-802a-c87aaed10ef7.png)

Mobile
![image](https://user-images.githubusercontent.com/219295/107168902-31606d00-6982-11eb-891d-53095a9b97c6.png)

**Reply by robertsLando** [reply]
I like it but where do you show nodes values etc? Everything that was on the expansion panel

**Reply by robertsLando** [reply]
I also like where you placed the controller status, I think that's the best place for that as it will be visible ion all panels

**Reply by jarrettv** [reply]
node values would go inside a dialog or slide out panel

**Reply by jarrettv** [reply]
![image](https://user-images.githubusercontent.com/219295/107241766-e893cd00-69f0-11eb-9826-25b424e1da2a.png)

```    getNodeColor(node) { 
      switch(node.status) {
        case 'Awake':
        case 'Alive': 
          return node.interviewStage === 'RestartFromCache' ? 'orange' : 'green'
        case 'Dead': 
          return 'red'
        case 'Asleep':
          return 'blue'
        default:
          return 'gray'
      }
    },
    getNodeIcon(node) {
      switch(node.interviewStage) {
        case 'None': 
          return 'remove_circle_outline'
        case 'RestartFromCache': 
          return node.status === 'Dead' ? 'dangerous' : 'history'
        case 'Complete': 
          return node.status === 'Asleep' ? 'nightlight_round' : 'check_circle_outline'
        case 'ProtocolInfo':
        case 'NodeInfo':
        case 'CommandClasses':
        case 'Neighbors':
          return 'arrow_circle_down'
        default:
          return node.status === 'Unknown' ? 'help_outline' : 'error_outline'
      }
    },```

**Reply by robertsLando** [reply]
@ahochsteger Thoughts?

**Comment by jarrettv** [comment]
Goal: remove broadcast question and remove node selection. Have a top level menu that affects all nodes and have an individual node menu that targets a specific node. For example:

**Controller advanced** (top level):
- Heal network
- Hard reset
- Refresh all node values (broadcast)
- Re-interview all nodes (broadcast)
- Check for failed nodes (broadcast)
- Remove all failed nodes (broadcast)
- Remove all associations (broadcast)

**Node advanced** (target individual node)
- Refresh node values
- Re-interview
- Heal
- Replace failed
- Remove failed
- Firmware update
- Remove all associations

**Reply by jarrettv** [reply]
![image](https://user-images.githubusercontent.com/219295/107301646-7b5c5800-6a41-11eb-84a8-fb0af11128cf.png)

**Comment by jarrettv** [comment]
![image](https://user-images.githubusercontent.com/219295/107312812-e1070f00-6a56-11eb-90ac-dce26587f331.png)
i've got most of this working on my branch although I am having an issue with `Import/Export`

**Reply by robertsLando** [reply]
I really like it! I need to know where all the other stuff is gone. I would also know @ahochsteger opinion as he is working on the nodes table too

**Reply by ahochsteger** [reply]
> I really like it! I need to know where all the other stuff is gone. I would also know @ahochsteger opinion as he is working on the nodes table too

Me too - see <https://github.com/zwave-js/zwavejs2mqtt/pull/524#issuecomment-776261294> for further comments.

**Comment by jarrettv** [comment]
New node dialog
![image](https://user-images.githubusercontent.com/219295/107700143-b0a0ba00-6c7c-11eb-9312-a0a3fa1078bc.png)

**Reply by robertsLando** [reply]
That's awesome 😍

**Comment by AlCalzone** [comment]
The info text for re-interview should be reworked IMO. It should make **very clear** that **all** currently known information about the node is reset and re-queried

**Comment by JonLaliberte** [comment]
Love the ideas here. Especially moving node details to a separate view (so not a part of the main list). Right now on mobile it's frustrating figuring out what you're looking at since the node details spill off the screen and it's easy to end up looking at the wrong node.

**Reply by robertsLando** [reply]
@JonLaliberte I liked the idea too. @jarrettv started with an implementation but unfortunately never finished it and now that should be rewritten almost all from scratch giving that most source code is changed

**Reply by JonLaliberte** [reply]
Yeah, this is a tough project since it's not as usually lived in (a UI/UX used daily). I don't have the skills to help here unfortunately, but I hope someone is able to pick this up at some point. Thanks to everyone involved!

**Comment by robertsLando** [comment]
Finally found some time to look at this, check https://github.com/zwave-js/zwave-js-ui/pull/2971

**Comment by mckochan** [comment]
Thanks to all! This makes inclusion on mobile much easier

**Reply by robertsLando** [reply]
Glad you like it, let me know if you have some improvements to suggest

**Reply by mckochan** [reply]
Mine looks a little bit different then https://github.com/zwave-js/zwave-js-ui/pull/2971
Looks like the Node# and status icon are in different spots
![Screenshot 2023-03-08 at 9 07 24 AM](https://user-images.githubusercontent.com/57327555/223735068-1f1a1abc-9bb6-4f08-b737-0a0f6d211954.png)
Instead of the big status icon (move/make smaller) and FW version (remove), I would put Manufacturer and product. Long term goal for this could be to import the product image from your device database like zigbee2mqtt does.
To use the horizontal space, either expand the tile to fit the width of the screen or keep square and allow 2 rows of tiles.

**Reply by robertsLando** [reply]
> Mine looks a little bit different then #2971

Yeah because that wasn't the last style, I made that video some commit before merging. All icons looks good to me, the only thing I see there is that you don't have any node name in the controller node so the space isn't filled very well in vertical alignment

> Instead of the big status icon (move/make smaller) and FW version (remove), I would put Manufacturer and product. Long term goal for this could be to import the product image from your device database like zigbee2mqtt does.

ATM I thought about using product image but that would require an internet connection and also wouldn't be always available, also manufacturer and other infos are already available in the tooltip and sometimes them could be very long (too long to diplay them in that little space

**Reply by JonLaliberte** [reply]
I’m a big fan of the update. 

I agree with @mckochan here:
> To use the horizontal space, either expand the tile to fit the width of the screen or keep square and allow 2 rows of tiles.

**Reply by robertsLando** [reply]
Done

![Schermata da 2023-03-08 17-12-28](https://user-images.githubusercontent.com/11502495/223767419-298a8b71-b34f-44db-b540-d8afc00e73eb.png)

**Reply by JonLaliberte** [reply]
Awesome!

**Reply by robertsLando** [reply]
To test it just use `master` tag, giving that I just release a new minor I will keep this for next release in few days
